src/pytest_nm_releng/plugin.py

```diff
@@ -15,18 +15,48 @@
 
 import os
 from datetime import datetime, timezone
+from enum import Enum
 from pathlib import Path
 
+from uuid_utils import uuid4, uuid7
 
-def get_utc_timestamp() -> float:
-    return datetime.now(timezone.utc).timestamp()
+
+class SuffixType(Enum):
+    TIMESTAMP = "timestamp"
+    UUID4 = "uuid4"
+    UUID7 = "uuid7"
+
+
+DEFAULT_SUFFIX_TYPE = SuffixType.TIMESTAMP
+
+
+def get_utc_timestamp() -> str:
+    return str(datetime.now(timezone.utc).timestamp())
+
+
+def generate_suffix(suffix_type: SuffixType) -> str:
+    suffix_map = {
+        SuffixType.TIMESTAMP: get_utc_timestamp,
+        SuffixType.UUID4: lambda: str(uuid4()),
+        SuffixType.UUID7: lambda: str(uuid7()),
+    }
+    return suffix_map.get(suffix_type, get_utc_timestamp)()
 
 
 def generate_junit_flags() -> list[str]:
     if not (junitxml_base_dir := os.getenv("NMRE_JUNIT_BASE")):
         return []
 
-    junitxml_file = Path(junitxml_base_dir) / f"{get_utc_timestamp()}.xml"
+    junit_suffix_type = os.getenv("NMRE_JUNIT_SUFFIX_TYPE", DEFAULT_SUFFIX_TYPE)
+    junit_suffix_type = (
+        SuffixType(junit_suffix_type)
+        if junit_suffix_type in SuffixType._value2member_map_
+        else SuffixType.TIMESTAMP
+    )
+
+    junitxml_file = (
+        Path(junitxml_base_dir) / f"{generate_suffix(junit_suffix_type)}.xml"
+    )
 
     if prefix := os.getenv("NMRE_JUNIT_PREFIX"):
         junitxml_file = junitxml_file.with_name(f"{prefix}{junitxml_file.name}")
```

Comment:
Should we apply "Path(junitxml_base_dir) /" as a last step after prefix is applied first?

Reply:
I like having the prefix after mainly to flow through the feature logic in a particular way (the prefix is an optional 'input' so we conditionally apply it after the pieces that are required). However, I can see the logic in moving it before the path creation and I think that would also eliminate a single line of code. I’ll think about this a bit and probably adopt the change.

Reply:
The main issue was that the prefix was applied before the base/suffix.xml but I see the updated code looks good now, thanks.

Reply:
Ah, yeah – the prefix has been an existing feature for a while and the way it was applied prior was not an issue with the way the `Path` methods work – basically, it was replacing the existing filename (just the filename and not any folder names before it) with the new filename that was constructed with the prefix + original filename.